include/openauto/Service/MediaStatusService.hpp

```diff
@@ -0,0 +1,52 @@
+/*
+*  This file is part of openauto project.
+*  Copyright (C) 2018 f1x.studio (Michal Szwaj)
+*
+*  openauto is free software: you can redistribute it and/or modify
+*  it under the terms of the GNU General Public License as published by
+*  the Free Software Foundation; either version 3 of the License, or
+*  (at your option) any later version.
+
+*  openauto is distributed in the hope that it will be useful,
+*  but WITHOUT ANY WARRANTY; without even the implied warranty of
+*  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+*  GNU General Public License for more details.
+*
+*  You should have received a copy of the GNU General Public License
+*  along with openauto. If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#pragma once
+
+#include "aasdk/Channel/AV/MediaStatusServiceChannel.hpp"
+#include "IService.hpp"
+
+namespace openauto
+{
+namespace service
+{
+class IAndroidAutoInterface;
+class MediaStatusService: public aasdk::channel::av::IMediaStatusServiceChannelEventHandler, public IService, public std::enable_shared_from_this<MediaStatusService>
+{
+public:
+    MediaStatusService(boost::asio::io_service& ioService, aasdk::messenger::IMessenger::Pointer messenger, IAndroidAutoInterface* aa_interface);
```

Comment:
the constructor here doesnt really need to be taking aa interface as a param here does it? looks like itll always be null unless explicitly set in the service factory

Reply:
let me double check how I've implemented this - I think I was worried about scenarios that don't exist

Reply:
but if its needed, id say add it to the nav interface to have that consistency (and just in case itd be useful in the future)

Reply:
Ok double checked this and the tl;dr is that it's _kinda_ needed.

services aren't actually initialized until an android device is connected (and create() is called in ServiceFactory).
So depending on when the androidautointerface is set by an external application (dash), it's either being set before or after service initialization, so by having it in the constructor as well as explicitly set, we handle both cases.